lib/trino-arrow/src/main/java/io/trino/arrow/type/TimeWithValueTimezoneType.java

```diff
@@ -0,0 +1,69 @@
+package io.trino.arrow.type;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.*;
+import org.apache.arrow.vector.complex.StructVector;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.FieldType;
+
+public class TimeWithValueTimezoneType extends ArrowType.ExtensionType {
```

Comment:
it seems like zoned time and timestamps in trino handle the zone at the row level (ie different rows can have different zones). arrow scopes the zone to the column.